### PR TITLE
[ROCm][Bugfix] Re-tag AITER MoE weights as preshuffled after replace_parameter

### PR DESCRIPTION
**Pull request by maeehart**
## Summary

In `Fp8MoEMethod`, `Mxfp4MoEMethod`, `GptOssMxfp4MoEMethod`, and `UnquantizedFusedMoEMethod`, the AITER MoE backends call `rocm_aiter_ops.shuffle_weights()` (FP8 / unquantized via `aiter.ops.shuffle.shuffle_weight`) or `rocm_aiter_ops.shuffle_weight_a16w4()` (MXFP4 BF16) to lay the MoE weights out for AITER's tuned 2-stage CK kernels.

The follow-up `replace_parameter(layer, "w13_weight", ...)` calls then wrap the shuffled tensors in fresh `nn.Parameter` instances, **which do not propagate the custom `is_shuffled = True` Python attribute** that the shuffle helper attaches to the inner tensor. As a result, AITER's runtime kernel selection reads `getattr(layer.w13_weight, "is_shuffled", False) -> False`, falls back to the non-tuned preshuffle-off (`Nswizzle0`) path, and emits

```
[aiter] [fused_moe] tuned config found for (...) but is_shuffled=False.
Tuned kernels are optimized for preshuffled weights (preshuffle_on).
Running with preshuffle_off may produce incorrect results.
```

once per MoE layer per worker. The Quark MoE method already handles this correctly (`vllm/model_executor/layers/quantization/quark/quark_moe.py:1286-1287`, which sets `layer.w{13,2}_weight.is_shuffled = True` after the equivalent `shuffle_weights()` + `replace_parameter`).

This PR re-tags the layer Parameters explicitly after the `replace_parameter` calls in the four affected MoE methods, bringing them in line with the Quark path.

## Files changed

- `vllm/model_executor/layers/quantization/fp8.py` — `Fp8MoEMethod` (1 site)
- `vllm/model_executor/layers/quantization/mxfp4.py` — `GptOssMxfp4MoEMethod._setup_kernel` and `Mxfp4MoEMethod._setup_kernel` (2 sites; both use the identical `_setup_kernel` block, so the patch is the same)
- `vllm/model_executor/layers/fused_moe/unquantized_fused_moe_method.py` — `UnquantizedFusedMoEMethod._setup_kernel` (1 site)

`AITER_MXFP4_FP8` is intentionally **excluded**: that backend uses Triton kernels (not AITER's 2-stage CK kernels) and does not go through the AITER `is_shuffled` check at runtime. The patch gates on `Mxfp4MoeBackend.AITER_MXFP4_BF16` specifically.

## Root cause walk-through (FP8 case, others analogous)

1. `vllm/model_executor/layers/fused_moe/oracle/fp8.py:445`:
   ```python
   elif fp8_backend == Fp8MoeBackend.AITER:
       w13, w2 = rocm_aiter_ops.shuffle_weights(w13, w2)
   ```
   `shuffle_weights()` returns tensors with `tensor.is_shuffled = True` (set in `aiter/ops/shuffle.py`).
2. `vllm/model_executor/layers/quantization/fp8.py:765`:
   ```python
   replace_parameter(layer, "w13_weight", w13)
   replace_parameter(layer, "w2_weight", w2)
   ```
   `replace_parameter` wraps the data in `nn.Parameter(w13, requires_grad=False)`. The new Parameter is a separate Python object; `Parameter.is_shuffled` raises `AttributeError`, so `getattr(..., False) -> False`.
3. At runtime, AITER's dispatcher (`aiter/fused_moe.py`) does
   ```python
   is_shuffled = getattr(w1, "is_shuffled", False)
   ...
   if not is_shuffled and not run_1stage:
       logger.warning(f"[fused_moe] tuned config found for {keys} but is_shuffled=False ...")
   ```
   The data IS in the preshuffled layout (we ran the shuffle in step 1), but AITER doesn't know that, so it picks the non-tuned `Nswizzle0` kernel variant.

## Test plan

Empirical verification on **DeepSeek-V3.2-Exp w8a8 block-scale FP8** running with `--tensor-parallel-size 4` on **MI355X**:

| | warnings per process | tuned 2-stage kernel selected |
|---|---:|---|
| before | **24** (`is_shuffled=False ... may produce incorrect results`) | `Nswizzle0` (preshuffle-off fallback) |
| after  | **0** | `Nswizzle1` (preshuffle-on, from `aiter/configs/model_configs/a8w8_blockscale_tuned_fmoe_ds_v3.csv`) |

`lm_eval` `gsm8k` (`num_concurrent=1, max_gen_toks=8192, --limit 24`) flexible-extract: 0.9583 (no regression vs. unpatched).

For MXFP4 / unquantized, the patch is structurally identical and verified by code review against the Quark precedent (`quark_moe.py:1286-1287`); empirical confirmation will follow as we run MXFP4-BF16 / unquantized MoE workloads through the same harness.

### CODE REVIEW

File: vllm/model_executor/layers/fused_moe/unquantized_fused_moe_method.py [hunk 1]
```diff
@@ -171,6 +171,18 @@ def _setup_kernel(
         replace_parameter(layer, "w13_weight", w13_new, prefer_copy=is_weight_update)
         replace_parameter(layer, "w2_weight", w2_new, prefer_copy=is_weight_update)
 
+        # AITER's shuffle_weights() (invoked inside convert_to_unquantized_kernel_format)
```
Comment (dllehr-amd) [inline vllm/model_executor/layers/fused_moe/unquantized_fused_moe_method.py]: Can we cut down on the comments here?  I think a simple AITER backend requires the weights to be shuffled would suffice

**Review by dllehr-amd — APPROVED** [review]
Looks good, but we don't need the large comments for a simple fix

File: vllm/model_executor/layers/fused_moe/unquantized_fused_moe_method.py (hunk 1 above)
Comment (maeehart) [reply vllm/model_executor/layers/fused_moe/unquantized_fused_moe_method.py]: Done — trimmed the comment at this site and the corresponding three other sites (FP8 1, MXFP4 2 sites, unquantized 1) to a single-line `# AITER backend requires weights to be marked as shuffled.` Force-pushed as a680c58.

**Review by tjtanaa — APPROVED** [review]
LGTM
